src/app/person-search/components/person-search-header/person-search-header.component.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { PersonSearchHeaderComponent } from './person-search-header.component';
+import { buildTestModuleMetadata } from '../../../../spec-helpers';
+import { Router } from '@angular/router';
+
+describe('PersonSearchHeaderComponent', () => {
+  let component: PersonSearchHeaderComponent;
+  let fixture: ComponentFixture<PersonSearchHeaderComponent>;
+
+  let element: HTMLElement;
+  let router: Router;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule(
+      buildTestModuleMetadata({ declarations: [PersonSearchHeaderComponent] })
+    ).compileComponents();
+
+    fixture = TestBed.createComponent(PersonSearchHeaderComponent);
+    component = fixture.componentInstance;
+    element = fixture.debugElement.nativeElement;
+    fixture.detectChanges();
+
+    router = TestBed.inject(Router);
+    spyOn(router, 'navigate');
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should build url', () => {
+    const key = 12;
+    component.navigateToDossier(key);
+    expect(router.navigate).toHaveBeenCalledWith([
+      'person-search',
+      'student',
+      key,
+      'addresses',
+    ]);
+  });
```

Comment:
Ein Detail: vom Verhalten («Behavior») her müsste die Spec eher heissen `it("navigates to dossier with the given id")` oder so...

Und wenn wir schon bei BDD-Testing sind – es wäre gut, wenn du mit `describe` «beschreibst» _was_ du testest und dann mit `it`s _wie_ sich das beschriebene _verhalten_ soll. Beispiel:

```
describe(".navigateToDossier", () => {
  it("navigates to dossier with the given id", () => {
    // ...
  });

  // Weitere `it`s falls es weitere Verhalten gibt...
});
```

Reply:
:+1: